models/demos/t3000/falcon40b/tests/test_perf_falcon.py

```diff
@@ -199,13 +208,26 @@ def run_test_FalconCausalLM_end_to_end(
             ]
         )
     elif llm_mode == "decode":
+        tt_inputs, tt_attention_mask_host = tt_FalconCausalLM.model_preprocessing(
+            llm_mode, model_input, kv_cache_len, num_input_tokens=kv_len
+        )
+        attention_mask_memconfig = model_config["ATTN_MASK_MEMCFG"]
+        num_max_tokens = nearest_32(kv_cache_len + 1)
+        if attention_mask_memconfig.is_sharded():
+            attn_mask_shard_shape = attention_mask_memconfig.shard_spec.shape
+            attn_mask_shard_shape[-1] = num_max_tokens
+            attention_mask_memconfig.shard_spec.shape = attn_mask_shard_shape
         tt_attention_mask = [
             tt_attention_mask_host[i].to(devices[i], attention_mask_memconfig) for i in range(len(devices))
         ]
```

Comment:
This is already done in model_preprocessing and the mask is already on device right?

Reply:
Yeah you're right! Thanks for pointing that out. I removed all the attention mask processing in the perf test file now, it was all redundant.